frontend/src/components/core/StreamlitDialog/StreamlitDialog.test.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Fragment } from "react"
+import { mount } from "lib/test_util"
+import { StreamlitDialog, DialogType } from "./StreamlitDialog"
+
+describe("StreamlitDialog", () => {
+  it("renders clear cache dialog and focuses clear cache button", () => {
+    const wrapper = mount(
+      <Fragment>
+        {StreamlitDialog({
+          type: DialogType.CLEAR_CACHE,
+          confirmCallback: () => {},
+          defaultAction: () => {},
+          onClose: () => {},
+        })}
+      </Fragment>
+    )
+
+    // Wait until the next frame of the event loop, to give React a chance to
```

Comment:
Hm, is there a chance that this might end up flaky since we don't have total control over what happens in the event loop?

Reply:
To be honest, not sure. I saw [this comment](https://github.com/facebook/react/issues/11851#issuecomment-351780994) by the React team that `autoFocus` calls `.focus()` on mount. This whole time I've just been mindlessly using this trick to wait for componentDidMount to run.

Looking online, though, apparently flushing Promises with `setImmediate` is the [way to go](https://github.com/testing-library/react-testing-library/issues/11), so I changed it.